lib/Service/StorageService.php

```diff
@@ -338,11 +338,7 @@ public function getAllFilesInFolder(Node $node): \Generator {
 		$filesGen = $this->getFilesInMount($mount->getNumericStorageId(), $node->getId(), 0, 0);
 
 		foreach ($filesGen as $fileId) {
-			$node = current($this->rootFolder->getById($fileId));
```

Comment:
This could get really expensive

Reply:
do we already filter our the `appdata_*/*` files, files_trashbin, etc. in the oc_filecache query? This might've been helping with that.